src/hidimstat/statistical_tools/gaussian_knockoffs.py

```diff
@@ -0,0 +1,186 @@
+import warnings
+
+import numpy as np
+from sklearn.utils import check_random_state
+
+
+class GaussianKnockoffs:
+    """
+    Generator for second-order Gaussian variables using the equi-correlated method.
+    Creates synthetic variables that preserve the covariance structure of the original
+    variables while ensuring conditional independence between the original and synthetic data.
+    Parameters
+    ----------
+    cov_estimator : object
+        Estimator for computing the covariance matrix. Must implement fit and
+        have a covariance_ attribute after fitting.
+    random_state : int or None, default=None
+        Random seed for generating synthetic data.
+    tol : float, default=1e-14
+        Tolerance for numerical stability in matrix computations.
+    Attributes
+    ----------
+    mu_tilde_ : ndarray of shape (n_samples, n_features)
+        Mean matrix for generating synthetic variables.
+    sigma_tilde_decompose_ : ndarray of shape (n_features, n_features)
+        Cholesky decomposition of the synthetic covariance matrix.
+    References
+    ----------
+    .. footbibliography::
+    """
+
+    def __init__(self, cov_estimator, random_state=None, tol=1e-14):
+        self.cov_estimator = cov_estimator
+        self.tol = tol
+        self.rng = check_random_state(random_state)
```

Comment:
Related to #360 to allow reproducibility, we should have in .__init__():
```self.random_state = random_state```
and in .sample()
```rng = check_random_state(random_state)```
```suggestion
        self.random_state = random_state
```